app/services/doc_auth/lexis_nexis/request.rb

```diff
@@ -0,0 +1,157 @@
+module DocAuth
+  module LexisNexis
+    class Request
+      attr_reader :config
+
+      def initialize(config:)
+        @config = config
+      end
+
+      def fetch
+        http_response = send_http_request
+        return handle_invalid_response(http_response) unless http_response.success?
+
+        handle_http_response(http_response)
+      rescue Faraday::ConnectionFailed, Faraday::TimeoutError => e
+        handle_connection_error(e)
+      end
+
+      def metric_name
+        raise NotImplementedError
+      end
+
+      private
+
+      def send_http_request
+        case method.downcase.to_sym
+        when :post
+          send_http_post_request
+        when :get
+          send_http_get_request
+        end
+      end
+
+      def handle_http_response(_response)
+        raise NotImplementedError
+      end
+
+      def handle_invalid_response(http_response)
+        message = [
+          self.class.name,
+          'Unexpected HTTP response',
+          http_response.status,
+        ].join(' ')
+        exception = DocAuth::RequestError.new(message, http_response.status)
+
+        handle_connection_error(exception)
+      end
+
+      def handle_connection_error(exception)
+        config.exception_notifier&.call(exception)
```

Comment:
if we want, we can remove this notion of `exception_notifier` entirely and switch to `NewRelic::Agent.notice_error`

Reply:
Done in b7b535a